src/sagemaker_xgboost_container/algorithm_mode/hyperparameter_validation.py

```diff
@@ -123,6 +123,7 @@ def interaction_constraints_validator(value, dependencies):
         hpv.IntegerHyperparameter(name="early_stopping_rounds", range=hpv.Interval(min_closed=1), required=False),
         hpv.CategoricalHyperparameter(name="booster", range=["gbtree", "gblinear", "dart"], required=False),
         hpv.IntegerHyperparameter(name="verbosity", range=hpv.Interval(min_closed=0, max_closed=3), required=False),
+        hpv.IntegerHyperparameter(name="nfold", range=hpv.Interval(min_closed=2), required=False, tunable=False),
```

Comment:
If `nfold` is internal for autopilot use, please mark it for internal use by renaming it to `_nfold`. See `_tuning_objective_metric`: https://github.com/aws/sagemaker-xgboost-container/blob/master/src/sagemaker_xgboost_container/algorithm_mode/hyperparameter_validation.py#L216.

With our lack of thorough documentation, customers often resort to code, so they will start using this. Most of them will just try it and see that it doesn't work as expected and won't report the issue. Those that do will be filtered incorrectly by SAs who again refer to code to give (incorrect) advice. If it finally reaches our team, our only explanation will be that it's meant for internal use.

I guess it's ultimately Rahul. Product, and/or your call with naming, but I feel pretty strongly about it.

Reply:
Although it is for autopilot for now, we may enable it for public usage in the future when we have requests from customers.

Reply:
Then we can change it back to `nfold` when we enable it for public usage. It should be marked for internal use if the current target is only autopilot.